android/app/src/main/java/com/mattermost/rnbeta/MattermostManagedModule.java

```diff
@@ -0,0 +1,70 @@
+package com.mattermost.rnbeta;
+
+import android.app.Application;
+import android.content.Context;
+import android.os.Bundle;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.NativeModule;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+
+public class MattermostManagedModule extends ReactContextBaseJavaModule {
+    public static MattermostManagedModule instance;
+
+    private boolean shouldBlurAppScreen = false;
+    private ReactApplicationContext reactContext;
+
+    private MattermostManagedModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+
+        this.reactContext = reactContext;
+    }
+
+    public static MattermostManagedModule getInstance(ReactApplicationContext reactContext) {
+        if (instance == null) {
+            instance = new MattermostManagedModule(reactContext);
+        }
+
+        return instance;
+    }
+
+    @Override
+    public String getName() {
+        return "MattermostManaged";
+    }
+
+    @ReactMethod
+    public void blurAppScreen(boolean enabled) {
+        this.shouldBlurAppScreen = enabled;
+    }
+
+    public boolean isBlurAppScreenEnabled() {
+        return this.shouldBlurAppScreen;
+    }
+
+    @ReactMethod
+    public void getConfig(final Promise promise) {
+        try {
+            Bundle config = MainApplication.instance.notificationsLifecycleFacade.getManagedConfig();
```

Comment:
Is it possible to pass the instance of `MainApplication` down into this object via the constructor? Static access of fields like this is often a bad idea, because it leaves open the possibility that `instance` hasn't yet been initialized, potentially causing a null ref exception. If instead, you pass it down into this object, then you'll know that it has always been set up correctly by the time you try to use it